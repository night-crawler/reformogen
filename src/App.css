.formogen .files-previews {
  max-height: 50vh;
  height: auto;
  overflow-x: hidden;
  overflow-y: auto;
  padding-right: 10px;
}
