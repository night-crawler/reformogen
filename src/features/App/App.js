import React, { Component } from 'react';
import { Segment, Grid, Button } from 'semantic-ui-react';
import { FormogenForm } from 'reformogen-redux/build/FormogenForm';

import { FormComponent } from '~/formogen/semantic-ui/FormComponent';

import { getFieldComponentForType } from 'reformogen-redux/build/semantic-ui';


export class App extends Component {
  state = { showFirst: true };
  
  render() {
    return (
      <Grid columns='3'>
        <Grid.Column>
          <Button content='Check clean up' fluid={ true } onClick={ () => {
            // eslint-disable-next-line          
            this.setState({ showFirst: !this.state.showFirst })
          } } />
          { this.state.showFirst && 
            <Segment color='red'>
              <FormogenForm
                formId='form-1'
                locale='ru'
                getFormComponent={ () => FormComponent }
                getFieldComponent={ getFieldComponentForType }
                describeUrl='http://localhost:8000/api/v1/sample/authors/describe/'
                createUrl='http://localhost:8000/api/v1/sample/authors/'
              />
            </Segment>
          }
        </Grid.Column>
        
        <Grid.Column>
          <Segment color='red'>
            <FormogenForm
              formId='form-2'
              getFormComponent={ () => FormComponent }
              getFieldComponent={ getFieldComponentForType }
              describeUrl='http://localhost:8000/api/v1/sample/authors/1/describe_object/'
              objectUrl='http://localhost:8000/api/v1/sample/authors/1/'
            />
          </Segment>
        </Grid.Column>

        <Grid.Column>
          <Segment color='red'>
            <FormogenForm
              formId='form-3'
              getFormComponent={ () => FormComponent }
              getFieldComponent={ getFieldComponentForType }
              describeUrl='http://localhost:8000/api/v1/sample/books/1/describe_object/'
              objectUrl='http://localhost:8000/api/v1/sample/books/1/'
            />
          </Segment>
        </Grid.Column>
      </Grid>
    );
  }
}
