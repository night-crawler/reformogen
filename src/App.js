import './formogen/components/semantic-ui/styles';
import 'moment/locale/ru';

import loglevel from 'loglevel';
import prefix from 'loglevel-plugin-prefix';

import { Segment } from 'semantic-ui-react';
import React, { Component } from 'react';

import Formogen from './formogen-react-redux';

import { Button, Form, Grid } from 'semantic-ui-react';
import 'semantic-ui-css/semantic.css';
import preparedMetaData from './form.json';

import './App.css';


// TODO: it should be in formogen module
window.loglevel = loglevel;
prefix.apply(loglevel, { template: '[%t] %l (%n)' });


function FormComponent(props) {
    // it's here as an example of formogen form redefinition via props
    const _props = {
        ...props,
    };
    return <Form { ..._props } />;
}

function SubmitComponent(props) {
    // it's here as an example of formogen submit redefinition via props
    const _props = {
        ...props,
        primary: true,
        content: 'Primary',
        fluid: false,
    };
    return <Button { ..._props } />;
}


class App extends Component {
    constructor(props) {
        super(props);

        this.log = loglevel.getLogger('App.js');
        this.log.debug('initialized');

        this.state = {
            // canRenderAuthorPhotosForms: false,
            // authorPhotosMetaDataUrl: null,

            authorPhotosMetaDataUrl: 'http://localhost:8000/api/v1/sample/author-photos/describe/',
            authorPhotosWithoutAuthorMetaDataUrl: 'http://localhost:8000/api/v1/sample/author-photos/describe_without_author/',
            authorPhotosCreateUrl: 'http://localhost:8000/api/v1/sample/author-photos/',
            canRenderAuthorPhotosForms: true,
            // authorId: formData.id,
        };
    }

    handleAuthorEditFormLoaded = (metaData, formData) => {
        if (!formData.id)
            return;

        this.setState({
            authorPhotosMetaDataUrl: 'http://localhost:8000/api/v1/sample/author-photos/describe/',
            authorPhotosWithoutAuthorMetaDataUrl: 'http://localhost:8000/api/v1/sample/author-photos/describe_without_author/',
            authorPhotosCreateUrl: 'http://localhost:8000/api/v1/sample/author-photos/',
            canRenderAuthorPhotosForms: true,
            authorId: formData.id,

        });
    };

    render() {
        const prepopulatedFormData = {
            assigned_test_field: 'hahahah!',

            name: 'Mr. Inconspicuous',
            // dt_birth: new Date(),
            // dt_death: new Date(),
            state: 30,
            vasya: 42,
        };

        // TODO: add lazy evaluation for modal mode
        // TODO: add X (cancel) to modal window
        // TODO: check undefined metaDataUrl
        return (
            <div className='App'>
                <Grid columns={ 3 } stackable={ true }>
                    <Grid.Column>
                        <Segment className='formogen'>


                            <Formogen
                                locale={ 'ru' }
                                showHeader={ true }
                                title={ 'default formId aka "default"' }

                                /* author photo form */
                                metaDataUrl={ 'http://localhost:8000/api/v1/sample/author-photos/describe/' }
                                objectCreateUrl={ 'http://localhost:8000/api/v1/sample/author-photos/' }
                                objectUrl={ 'http://localhost:8000/api/v1/sample/author-photos/1/' }

                                /* author form */
                                // metaDataUrl={ 'http://localhost:8000/api/v1/sample/authors/describe/' }
                                // objectCreateUrl={ 'http://localhost:8000/api/v1/sample/authors/' }
                                // objectUrl={ 'http://localhost:8000/api/v1/sample/authors/23/' }

                                /* misc */
                                sendFileQueueLength={ 3 }

                                /* modal opts */
                                // showAsModal={ true }
                                // modalComponent={ null }
                                // modalTriggerComponent={ <Button>Rise up evil!</Button> }
                                // modalProps={ { dimmer: 'blurring' } }

                                // formComponent={ FormComponent }
                                // submitComponent={ SubmitComponent }
                            />


                        </Segment>
                    </Grid.Column>


                    <Grid.Column>
                        <Segment className='formogen'>

                            <Formogen
                                formId='formogen-1'

                                locale={ 'ru' }
                                showHeader={ true }
                                title={ 'formId = "formogen-1"' }

                                /* author photo form */
                                // metaDataUrl={ 'http://localhost:8000/api/v1/sample/author-photos/describe/' }
                                // objectCreateUrl={ 'http://localhost:8000/api/v1/sample/author-photos/' }
                                // objectUrl={ 'http://localhost:8000/api/v1/sample/author-photos/100/' }

                                /* author form */
                                metaDataUrl={ 'http://localhost:8000/api/v1/sample/authors/describe/' }
                                objectCreateUrl={ 'http://localhost:8000/api/v1/sample/authors/' }
                                objectUrl={ 'http://localhost:8000/api/v1/sample/authors/1/' }

                                /* misc */
                                sendFileQueueLength={ 3 }

                                /* modal opts */
                                // showAsModal={ true }
                                // modalComponent={ null }
                                // modalTriggerComponent={ <Button>Rise up evil!</Button> }
                                // modalProps={ { dimmer: 'blurring' } }

                                // formComponent={ FormComponent }
                                // submitComponent={ SubmitComponent }
                            />


                        </Segment>
                    </Grid.Column>
                </Grid>
            </div>
        );
    }
}

export default App;
