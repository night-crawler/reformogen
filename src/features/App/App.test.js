
import { App } from './App';

it('renders without crashing', () => {
  const wrapper = shallow(<App />).dive();
  expect(wrapper).toMatchSnapshot();
});
