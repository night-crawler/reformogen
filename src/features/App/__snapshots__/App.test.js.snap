// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid
    columns="2"
  >
    <GridColumn>
      <Segment
        color="red"
      >
        <Connect(FormogenForm)
          describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
          formId="bla"
          getFieldComponent={[Function]}
          getFormComponent={[Function]}
        />
      </Segment>
    </GridColumn>
  </Grid>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GridColumn>
        <Segment
          color="red"
        >
          <Connect(FormogenForm)
            describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
            formId="bla"
            getFieldComponent={[Function]}
            getFormComponent={[Function]}
          />
        </Segment>
      </GridColumn>,
      "className": "ui two column grid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Segment
          color="red"
        >
          <Connect(FormogenForm)
            describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
            formId="bla"
            getFieldComponent={[Function]}
            getFormComponent={[Function]}
          />
        </Segment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Connect(FormogenForm)
            describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
            formId="bla"
            getFieldComponent={[Function]}
            getFormComponent={[Function]}
          />,
          "color": "red",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "describeUrl": "http://localhost:8000/api/v1/sample/authors/describe/",
            "formId": "bla",
            "getFieldComponent": [Function],
            "getFormComponent": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GridColumn>
          <Segment
            color="red"
          >
            <Connect(FormogenForm)
              describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
              formId="bla"
              getFieldComponent={[Function]}
              getFormComponent={[Function]}
            />
          </Segment>
        </GridColumn>,
        "className": "ui two column grid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Segment
            color="red"
          >
            <Connect(FormogenForm)
              describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
              formId="bla"
              getFieldComponent={[Function]}
              getFormComponent={[Function]}
            />
          </Segment>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(FormogenForm)
              describeUrl="http://localhost:8000/api/v1/sample/authors/describe/"
              formId="bla"
              getFieldComponent={[Function]}
              getFormComponent={[Function]}
            />,
            "color": "red",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "describeUrl": "http://localhost:8000/api/v1/sample/authors/describe/",
              "formId": "bla",
              "getFieldComponent": [Function],
              "getFormComponent": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
